### PR TITLE
New dihedral optimization scheme + MM minimize

### PR DESCRIPTION
**Pull request by j3mdamas**
This is the first approach at the new dihedral optimization scheme using MM minimization

Not to merge yet. Missing the analogous MM for the initial minimization.

Things to do:

- [x] Minimization with MM

### CODE REVIEW

**Comment by j3mdamas** [comment]
Not finished yet, just changes reviewer to @stefdoerr

File: htmd/qm/custom.py [hunk 1]
```diff
@@ -97,6 +97,8 @@ def minimize(self, coords, restrained_dihedrals):
 
             for dihedral in restrained_dihedrals:
                 theta0 = dihedralAngle(coords[dihedral])
+                # TODO: This is a work-around. jit signatures should be fixed instead so theta0 is never a ndarray
```
Comment (j3mdamas) [inline htmd/qm/custom.py]: @stefdoerr can you look at this work-around and see if the jit signatures can solve it? I don't know if you remember it.

Comment (stefdoerr) [reply htmd/qm/custom.py]: works fine on loro so I'm removing the work-around on my own branch to see if I catch it

File: htmd/parameterization/util.py [hunk 2]
```diff
@@ -268,3 +268,95 @@ def fitDihedrals(mol, qm, method, prm, all_dihedrals, dihedrals, outdir, geomopt
     return prm
 
 
+def fitDihedralsNew(mol, qm, method, prm, before_invention, all_dihedrals, dihedrals, outdir):
```
Comment (raimis) [inline htmd/parameterization/util.py]: Maybe `fitDihedralsNew` --> `fitDihedralsMM` or something more descriptive

**Comment by stefdoerr** [comment]
Leave this PR. It won't be merged. It's just temporary for my current work.
We will do it in a different way without code duplication etc

**Comment by stefdoerr** [comment]
@raimis don't merge this PR please. The minimizer is broken. I'll close it and make a new one after publication

**Pull request closed without merging**